internal/provider/resource_silo.go

```diff
@@ -0,0 +1,480 @@
+					},
+				},
+			},
+			"tls_certificates": schema.ListNestedAttribute{
+				Required:    true,
+				Description: "Initial TLS certificates to be used for the new Silo's console and API endpoints.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"cert": schema.StringAttribute{
+							Description: "PEM-formatted string containing public certificate chain.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"description": schema.StringAttribute{
+							Description: "Certificate description.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"key": schema.StringAttribute{
+							Description: "PEM-formatted string containing private key.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+						"name": schema.StringAttribute{
+							Description: "Name associated to the certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+							Validators: []validator.String{
+								stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
+									"allowed characters are lowercase ASCII, digits, and \"-\""),
+							},
+						},
+						"service": schema.StringAttribute{
+							Description: "Service using this certificate.",
+							Required:    true,
+							PlanModifiers: []planmodifier.String{
+								stringplanmodifier.RequiresReplace(),
+							},
+						},
+					},
+				},
+				Validators: []validator.List{
+					listvalidator.SizeAtLeast(1),
+				},
+			},
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Read:   true,
+				Update: true,
+				Delete: true,
+			}),
+			"time_created": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this silo was created.",
+			},
+			"time_modified": schema.StringAttribute{
+				Computed:    true,
+				Description: "Timestamp of when this silo was last modified.",
+			},
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "Unique, immutable, system-controlled identifier of the silo.",
+			},
+			"identity_mode": schema.StringAttribute{
+				Description: "How users and groups are managed in the Silo",
+				Computed:    true,
+				Default:     stringdefault.StaticString(string(oxide.SiloIdentityModeLocalOnly)),
+				Validators: []validator.String{
+					stringvalidator.OneOf(
+						string(oxide.SiloIdentityModeLocalOnly),
+						string(oxide.SiloIdentityModeSamlJit),
+					),
+				},
+			},
+		},
+	}
+}
+
+func siloCreateMappedFleetRolesModel(mappedFleetRoles map[string][]string) map[string][]oxide.FleetRole {
```

Comment:
At first I wasn't sure what this function was for but now I see this and its complement are here to convert to/from Terraform/Oxide types. Perhaps we can rename them to better suit their function.

```suggestion
func toOxideMappedFleetRoles(mappedFleetRoles map[string][]string) map[string][]oxide.FleetRole {
```